.github/workflows/cd.yml

```diff
@@ -0,0 +1,121 @@
+
+jobs:
+  build-server:
+    name: 'spendr server continuous deployment build'
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: '19'
+
+    - name: Install Dependencies
+      run: cd server && npm install
+
+    - name: Build Server
+      run: cd server && npm run build
+
+  build-client:
+    name: 'spendr client continuous deployment build'
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v3
+
+    - name: Set up Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: '19'
+
+    - name: Install Dependencies
+      run: cd client && npm install
+
+    - name: Build Client
+      run: cd client && npm run build
+
+    - name: Upload production ready build files
+      uses: actions/upload-artifact@v3
+      with:
+        name: spendr-client-static
+        path: ./client/build
+
+  publish-to-docker-server:
+    name: 'spendr server continuous deployment - publish'
+    needs: build-server
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/ci-cd-configurations'
+
+    steps:
+    - name: checkout code
+      uses: actions/checkout@v3
+
+    - name: Log on to docker hub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+
+    - name: Extract metadata (tags, labels) for Docker
+      id: meta
+      uses: docker/metadata-action@v3
+      with:
+        images: lysackm/spendr
+
+    - name: Build and push
+      uses: docker/build-push-action@v3
+      with:
+        file: ./.docker/server/Dockerfile
+        context: ./server
+        push: true
+        tags: ${{ steps.meta.outputs.tags }}
+        labels: ${{ steps.meta.outputs.labels }}
+    
+  publish-to-docker-client:
+    name: 'spendr client continuous deployment - publish'
+    needs: build-client
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/ci-cd-configurations'
+
+    steps:
+    - name: checkout code
+      uses: actions/checkout@v3
+
+    - name: Download artifact
+      uses: actions/download-artifact@v2
+      with:
+        name: spendr-client-static
+        path: ./.docker/client/static
+
+    - name: Log on to docker hub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+
+    - name: Extract metadata (tags, labels) for Docker
+      id: meta
+      uses: docker/metadata-action@v3
+      with:
+        images: lysackm/spendr
+        tags: |
+            type=raw,value=latest
+            type=sha
+
+    - name: Build and push
+      uses: docker/build-push-action@v3
+      with:
+        file: ./.docker/client/Dockerfile
+        context: .
+        push: true
+        tags: ${{ steps.meta.outputs.tags }}
+        labels: ${{ steps.meta.outputs.labels }}
```

Comment:
Out of scope: In the future it would be cool if we could get distinct tags for client and server images, or even versioning